front/src/applications/editor/tools/switchEdition/useSwitch.ts

```diff
@@ -0,0 +1,65 @@
+import { useContext, useMemo } from 'react';
+import { useSelector } from 'react-redux';
+import { first, keyBy } from 'lodash';
+import { SwitchEntity } from 'types';
+import EditorContext from 'applications/editor/context';
+import { getInfraID } from 'reducers/osrdconf/selectors';
+import { NEW_ENTITY_ID } from 'applications/editor/data/utils';
+import { SwitchEditionState, useSwitchTypes } from './types';
+import { ExtendedEditorContextType } from '../editorContextTypes';
+import { getSwitchTypeJSONSchema, switchToFlatSwitch } from './utils';
+
+// TODO : Rename all switch by tracknode when back renaming PR merged
+const useSwitch = () => {
+  const { state, editorState } = useContext(
+    EditorContext
+  ) as ExtendedEditorContextType<SwitchEditionState>;
+  // Retrieve base JSON schema:
```

Comment:
Insert some blank lines for more readability
```suggestion
  ) as ExtendedEditorContextType<SwitchEditionState>;
  
  // Retrieve base JSON schema:
```

Reply:
Done